x-pack/plugins/apm/public/components/app/Settings/anomaly_detection/legacy_jobs_callout.tsx

```diff
@@ -5,12 +5,41 @@
  */
 
 import { EuiCallOut, EuiButton } from '@elastic/eui';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { i18n } from '@kbn/i18n';
 import { useApmPluginContext } from '../../../../hooks/useApmPluginContext';
 
 export function LegacyJobsCallout() {
-  const { core } = useApmPluginContext();
+  const {
+    core,
+    plugins: { ml },
+  } = useApmPluginContext();
+
+  const [mlADLink, setMlADLink] = useState(
+    core.http.basePath.prepend('/app/ml/jobs')
+  );
+
+  useEffect(() => {
+    let isCancelled = false;
+    const generateLink = async () => {
+      if (ml?.urlGenerator !== undefined) {
+        const href = await ml.urlGenerator.createUrl({
+          page: 'jobs',
+          pageState: {
+            jobId: 'high_mean_response_time',
+          },
+        });
+        if (!isCancelled) {
+          setMlADLink(href);
+        }
+      }
+    };
+    generateLink();
+    return () => {
+      isCancelled = true;
+    };
+  }, [ml?.urlGenerator]);
+
```

Comment:
Requiring state and side effects to generate a static url seems like a big overkill of boilerplate from a consumers point of view. 

Suggestions for an api:
```ts
const mlLink = getMlLink({
  page: 'jobs',
  pageState: {
    jobId: 'high_mean_response_time'
  },
})
```

In the above case I'd expect the arguments to be strongly typed so I can't specify `page: 'this-does-not-exist'`

Reply:
I think url generators are useful for stateful urls that are to be persisted (in ES for instance) and thus need to be resilient across version upgrades. In this case we simply need a static url used at runtime. A synchronous function like the following would be vastly superior due to it's type safety and simplicity:

```ts
function getMlLink({page, pageState} : {page: Page, pageState: PageState}) {
  //... ml routing logic
 return mlLink;
}
```

Reply:
I'll resolve this comment since we already discussed this previously. I'll use the `useMlHref` hook for now for APM and work on a synchronous iteration for 7.11. Thanks.

Reply:
Sounds great. Thanks for doing this @qn895 !